custom_components/imou_life/__init__.py

```
@@ -197,33 +198,68 @@ def _configure_device_options(device: ImouDevice, entry: ConfigEntry):
         _LOGGER.debug("Initializing device with timeout %d seconds...", setup_timeout)
         await asyncio.wait_for(device.async_initialize(), timeout=setup_timeout)
         _LOGGER.debug("Device initialization completed")
+
+        # Clear rate limit state on successful initialization
+        rate_limit_mgr.clear_rate_limit(app_id, app_secret)
+
     except asyncio.TimeoutError:
         _LOGGER.error("Device initialization timed out after %d seconds", setup_timeout)
         raise ConfigEntryNotReady(
             translation_domain=DOMAIN, translation_key="device_init_timeout"
         )
```

Ruff (0.15.13)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)